internal/auth/yaml_storage.go

```diff
@@ -47,10 +59,25 @@ func (y *YamlStorage) WriteCalendarAuth(newCal CalendarAuth) (bool, error) {
 		return false, err
 	}
 
+	// Adding freshly written CalendarAuth to memory
+	// Probably unneeded, the next time this data will be retrieved is on the next calendarsync run
+	log.Debugf("Adding calendar auth for cal %s to memory", newCal.CalendarID)
+	y.DecryptedAuth = append(y.DecryptedAuth, newCal)
```

Comment:
I'd simplify this to `y.DecryptedAuth = cals`. That trivially ensures that the cached data and the on-disk state are equivalent.

Reply:
fixed, good catch!